service/storage_proxy.cc

```diff
@@ -5339,7 +5339,7 @@ future<> storage_proxy::wait_for_hints_to_be_replayed(utils::UUID operation_id,
         // Step 2: Wait until hints are replayed
         // Sleep 1 second between checks
         bool reached_sync_point = false;
-        while (!reached_sync_point && lowres_clock::now() < deadline) {
```

Comment:
I see you addressed the shutdown thing. What's the timeout of the rpc calls?

Reply:
It's equal to `deadline` - so, the timeout of the whole "wait for hints to be replayed" operation. RPC handlers for those verbs are short and should finish very quickly.